compiler/util/kernel_type.py

```diff
@@ -0,0 +1,248 @@
+from enum import Enum
+
+from xdsl.dialects import arith, linalg
+from xdsl.dialects.builtin import ShapedType
+from xdsl.ir import Block, Operation, OpResult, SSAValue
+
+
+class KernelException(Exception):
+    """
+    A KernelException is raised by the parsing functions if the provided
+    operation does not match the expected operation type in the kernel.
+    """
+
+    pass
+
+
+class InputException(KernelException):
+    """
+    A InputException is a specific case of parsing error, when we fail to
+    match given operands to the inputs of the kernel. The current parsing
+    strategy is therefore not necessarily incorrect, just incomplete.
+    """
+
+    pass
+
+
+class KernelType(Enum):
+    MUL = "mul"
+    MAC = "mac"
+    QMAC = "qmac"
```

Comment:
Add a docstring to this class, and a short comment on what each specific type is?
I don't remember exactly what you mean with a qmac and a mul operation tbh :shrug: